nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/main/java/org/apache/nifi/processors/salesforce/QuerySalesforceObject.java

```diff
@@ -261,6 +321,23 @@ protected Collection<ValidationResult> customValidate(ValidationContext validati
 
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        boolean isCustomQuery = CUSTOM_QUERY.getValue().equals(context.getProperty(QUERY_TYPE).getValue());
+        AtomicReference<String> nextRecordsUrl = new AtomicReference<>();
+        AtomicReference<String> totalSize = new AtomicReference<>();
```

Comment:
These `AtomicReference` objects are not used here. I don't think their creation should be the responsibility of `onTrigger`, rather that of the downstream methods.

The `nextRecordsUrl` constructor would be explicitly called both in those methods but I wouldn't consider that a duplication. These being closer to the `do...while` block they are used in would help understand their purpose much better.